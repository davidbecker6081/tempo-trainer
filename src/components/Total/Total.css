.total {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 15px;
  width: 25%;
  background-color: #252525;
  border-radius: 5px;
  box-shadow: 0px 0px 2px 2px #25252563;
  color: #FFFFFF;
  font-size: 1.3em;
}

.total h4,
.total p {
  margin: 10px 0;
}

.total h4 {
  font-weight: 300;
}

.total p {
  font-size: 1.2em;
  font-weight: 700;
}
