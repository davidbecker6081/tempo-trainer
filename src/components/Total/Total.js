import React from 'react';
import './Total.css';

const Total = ({ channelSet, dataHelper }) => {
  const lastEntry = dataHelper.data.samples[dataHelper.data.samples.length - 1];
  const totalMilli = lastEntry.millisecondOffset;
  const totalCleaner = {
    millisecondOffset: Math.round(dataHelper.convertMilliToMin(totalMilli) * 10) / 10,
    distance: Math.round((lastEntry.values.distance / 5280) * 100) / 100,
    elevation: Math.round(dataHelper.calculateTotalElevationGain()),
  };
  const channelCleaner = {
    millisecondOffset: 'time',
    distance: 'distance',
    elevation: 'elevation gain',
  };
  const descriptors = {
    millisecondOffset: 'minutes',
    distance: 'miles',
    elevation: 'ft',
  };

  return (
    <article>
      <h4>
        {channelCleaner[channelSet].toUpperCase()}
      </h4>
      <p>
        {totalCleaner[channelSet]} <span>{descriptors[channelSet]}</span>
      </p>
    </article>
  );
};

export default Total;
