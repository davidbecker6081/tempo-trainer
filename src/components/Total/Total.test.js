import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Total from './Total';
import DataHelper from '../DataCleaner/DataCleaner';
import workoutData from '../../__mock__/workout-data.json';

describe('Total', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      channelSet: '',
    };
  });

  describe('channelSet is defined', () => {
    beforeEach(() => { mockProps.channelSet = 'power'; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('dataHelper is defined', () => {
    beforeEach(() => { mockProps.dataHelper = new DataHelper(workoutData); });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
