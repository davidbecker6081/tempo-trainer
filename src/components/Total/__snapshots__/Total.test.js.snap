// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Total channelSet is defined should render correclty 1`] = `undefined`;

exports[`Total dataHelper is defined should render correclty 1`] = `undefined`;
